core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
+            if (metricsIdxCache.getIfPresent(metric.getMetricId()) == null) {
+                metricsIdxCache.put(metric.getMetricId(), true);
```

Comment:
Where do you propose to push this functionality? The data access layer is clearly not good because that is just queries; and the REST layer is just the glue code for the container. So only the core services is left. 

Also, how else can this be done other than filtering out metrics that do not need metrics index updates before passing them to the data access layer?